pkg/bmc/access.go

```diff
@@ -62,6 +67,9 @@ type AccessDetails interface {
 
 	// Whether the driver supports changing secure boot state.
 	SupportsSecureBoot() bool
+
+	// Build bios clean steps for ironic
+	BuildBIOSCleanSteps(firmwareConfig *metal3v1alpha1.FirmwareConfig) ([]nodes.CleanStep, error)
```

Comment:
It would certainly be easier to implement metal3-io/metal3-docs#173 if instead of returning CleanSteps here we just returned the `settings` to be passed to `apply_configuration`.
We're returning CleanSteps in order to implement the ResetSettings option, but I am not convinced that this option should exist (see previous comment).

Reply:
Done.